cpp/oneapi/dal/table/backend/convert.cpp

```diff
@@ -227,33 +227,50 @@ sycl::event convert_vector_device2host(sycl::queue& q,
     // To perform conversion, we gather data from device to host in temporary
     // contigious array and then run host conversion function
 
-    const std::int64_t element_size_in_bytes = dal::detail::get_data_type_size(src_type);
+    const std::int64_t dst_element_size_in_bytes = dal::detail::get_data_type_size(dst_type);
```

Comment:
collect more information about the destination to see if a shortcut can be used.

Reply:
Exactly!